okhttp/src/jvmMain/kotlin/okhttp3/Dispatcher.kt

```diff
@@ -111,6 +134,15 @@ class Dispatcher() {
     this.executorServiceOrNull = executorService
   }
 
+  constructor(asyncCallDispatcher: AsyncCallDispatcher) : this() {
```

Comment:
rather than this constructor and the next, how about a single one with default null param values?  This pattern doesn't scale when we add a 3rd or 4th constructor param.